app/controllers/vm_common.rb

```diff
@@ -1370,7 +1370,7 @@ def console_before_task(console_type)
   # Task complete, show error or launch console using VNC/MKS/VMRC task info
   def console_after_task(console_type)
     miq_task = MiqTask.find(params[:task_id])
-    if miq_task.status == "Error" || miq_task.task_results.blank?
+    unless miq_task.results_ready?
```

Comment:
@isimluk I was thinking more of creating "NOT Ok task status" and checking that task.results_ready? is false... basically for the conditionals previously there... where it makes sense.  It's really just unit tests for this new method.